test/module/shared_model/builders/protobuf/transport_builder_test.cpp

```diff
@@ -275,3 +283,88 @@ TEST_F(TransportBuilderTest, EmptyBlockCreationTest) {
       },
       [](const Error<std::string> &) { FAIL(); });
 }
+
+/**
+ * @given Valid block protobuf object with no transactions
+ * @when TransportBuilder tries to build BlockVariantType object
+ * @then built object contains EmptyBlock shared model object
+ * AND it is equal to the original object
+ */
+TEST_F(TransportBuilderTest, BlockVariantWithValidEmptyBlock) {
+  auto emptyBlock = createEmptyBlock();
+  interface::BlockVariantType orig_model =
+      std::make_shared<decltype(emptyBlock)>(emptyBlock.getTransport());
+
+  TransportBuilder<interface::BlockVariantType,
+                   validation::DefaultAnyBlockValidator>()
+      .build(emptyBlock.getTransport())
+      .match(
+          [&emptyBlock](const Value<decltype(orig_model)> &model) {
+            iroha::visit_in_place(
+                model.value,
+                [&emptyBlock](
+                    const std::shared_ptr<shared_model::interface::EmptyBlock>
+                        block) { EXPECT_EQ(emptyBlock, *block); },
+                [](const std::shared_ptr<shared_model::interface::Block>) {
+                  FAIL();
+                });
+          },
+          [](const Error<std::string> &) { FAIL(); });
+}
+
+/**
+ * @given Invalid block protobuf object with no transactions
+ * @when TransportBuilder tries to build BlockVariantType object
+ * @then build fails
+ */
+TEST_F(TransportBuilderTest, BlockVariantWithInvalidEmptyBlock) {
+  auto emptyBlock = createInvalidEmptyBlock();
+
+  TransportBuilder<interface::BlockVariantType,
+                   validation::DefaultAnyBlockValidator>()
+      .build(emptyBlock.getTransport())
+      .match([](const auto &) { FAIL(); },
```

Comment:
There are shortcut methods in result_fixture.hpp for this kind of checks. Please use them